### PR TITLE
Users By Native Language

### PR DESCRIPTION
**Pull request by yosefalnajjarofficial**
- Wrote the middleware for sending the JSON response of all users
- Wrote the query for getting all users who matches the native language id
- Wrote the tests for both the route and query
**Note**: I used the same format function and maybe I will write this code in a better way when we have time to do so 

Relates: #32

### CODE REVIEW

**Comment by Amoodaa** [comment]
`DISTINCT` is news for us, it will probably come handy soon
we are almost finishing up the backend
chat is left in next week
will be a lit week 🔥

File: test/queries/getUsersByLang.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    username: 'amoodaa',
+    bio: null,
+    avatar_path: null,
+    native_lang_id: 1,
+    learning_lang_id: 6,
+  };
+  dbBuild()
+    .then(() => getUsersByLang(1))
+    .then(({ rows: [actual] }) => {
+      t.deepEqual(actual, expected, 'The first user should be fadi');
```
Comment (fadeomar) [inline test/queries/getUsersByLang.test.js]:
```suggestion
      t.deepEqual(actual, expected, 'The first user should be amoodaa');
```

File: server/controllers/users/getUsersByLang.js [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+exports.getUsersByLang = (req, res, next) => {
+  let users;
+  const { id } = req.params;
+
+  if (!Number(id)) return next({ code: 400, msg: 'Bad request' });
+
+  getUsersByLang(id)
+    .then(({ rows }) => {
+      users = rows;
+      return users.map((user) => getLanguageById(user.native_lang_id));
```
Comment (11janci) [inline server/controllers/users/getUsersByLang.js]:
```suggestion
return Promise.all(
    users.map((user) => getLanguageById(user.native_lang_id))
);
```
then you can get rid of the below line

File: server/controllers/users/getUsersByLang.js [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+    })
+    .then((promises) => Promise.all(promises))
+    .then((langs) => formatLanguages(users, langs, 'native_lang_id'))
+    .then((result) => {
+      users = result;
+      return result.map((user) => getLanguageById(user.learning_lang_id));
```
Comment (11janci) [inline server/controllers/users/getUsersByLang.js]: same as above, return `Promise.all`

**Pull request merged**